File: Java/src/main/java/Solution801_900/Solution892.java
package Solution801_900;

/**
 * @author windows
 */
public class Solution892 {
    public int surfaceArea(int[][] grid) {
        int sum = 0;
        for (int i = 0; i < grid.length; i++) {
            int rowMax = 0;
            int colMax = 0;
            for (int j = 0; j < grid[i].length; j++) {
                rowMax = Math.max(rowMax, grid[i][j]);
                colMax = Math.max(colMax, grid[j][i]);
                if (grid[i][j] != 0){
                    sum += 2;
                }else{


                }
            }
            sum += rowMax * 2 + colMax * 2;
        }
        return sum;
    }



    public static void main(String[] args) {
        Solution892 solution892 = new Solution892();
    }
}

File: C++/src/Solution801_900/Solution892.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution892.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int Solution::surfaceArea(vector<vector<int>> &grid) {
    int n = grid.size();
    int res = 0;
    vector<vector<int>> dirs = {{1,  0},
                                {-1, 0},
                                {0,  1},
                                {0,  -1}};
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < n; ++j) {
            if (!grid[i][j]) {
                res += 2;
                for (int k = 0; k < dirs.size(); ++k) {
                    int x = i + dirs[k][0], y = j + dirs[k][1];
                    int val = 0;
                    if (x >= 0 && y >= 0 && x < n && y < n) {
                        val = grid[x][y];
                    }
                    res += max(grid[i][j] - val, 0);
                }
            }
        }
    }
    return res;
}

